ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -93,31 +93,51 @@ open Storage_interface
 open Listext
 open Fun
 
-let assert_sr_support_migration ~__context ~vdi_map ~remote =
+let assert_sr_support_operations ~__context ~vdi_map ~remote ~ops ~_type =
+  let op_supported_on_source_sr vdi ops =
+    (* Check VDIs must not be present on SR which doesn't have required capability *)
+    let source_sr = Db.VDI.get_SR ~__context ~self:vdi in
+    let sr_record = Db.SR.get_record_internal ~__context ~self:source_sr in
+    let sr_features = Xapi_sr_operations.features_of_sr ~__context sr_record in
+    List.iter (fun op ->
```

Comment:
This could be also implemented with `if not (List.for_all Smint.(has_capability op sr_features) ops) then raise … else ()`.  I think it would be a little clearer because it expresses the expectation that all `ops` must meet a requirement, otherwise an exception is thrown. There is a similar situation below.